// @alicloud/ocr20191230/dist/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
const oss_client_1 = __importStar(require("@alicloud/oss-client")), $OSS = oss_client_1;
const openplatform20191219_1 = __importStar(require("@alicloud/openplatform20191219")), $OpenPlatform = openplatform20191219_1;
const rpc_util_1 = __importDefault(require("@alicloud/rpc-util"));
const rpc_client_1 = __importStar(require("@alicloud/rpc-client")), $RPC = rpc_client_1;
const $OSSUtil = __importStar(require("@alicloud/oss-util"));
const tea_util_1 = __importDefault(require("@alicloud/tea-util"));
const $FileForm = __importStar(require("@alicloud/tea-fileform"));
const endpoint_util_1 = __importDefault(require("@alicloud/endpoint-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class RecognizePoiNameRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURL: 'ImageURL',
        };
    }
    static types() {
        return {
            imageURL: 'string',
        };
    }
}
exports.RecognizePoiNameRequest = RecognizePoiNameRequest;
class RecognizePoiNameResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: RecognizePoiNameResponseData,
        };
    }
}
exports.RecognizePoiNameResponse = RecognizePoiNameResponse;
class RecognizePoiNameAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURLObject: 'ImageURLObject',
        };
    }
    static types() {
        return {
            imageURLObject: 'Readable',
        };
    }
}
exports.RecognizePoiNameAdvanceRequest = RecognizePoiNameAdvanceRequest;
class DetectCardScreenshotRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURL: 'ImageURL',
        };
    }
    static types() {
        return {
            imageURL: 'string',
        };
    }
}
exports.DetectCardScreenshotRequest = DetectCardScreenshotRequest;
class DetectCardScreenshotResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: DetectCardScreenshotResponseData,
        };
    }
}
exports.DetectCardScreenshotResponse = DetectCardScreenshotResponse;
class DetectCardScreenshotAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURLObject: 'ImageURLObject',
        };
    }
    static types() {
        return {
            imageURLObject: 'Readable',
        };
    }
}
exports.DetectCardScreenshotAdvanceRequest = DetectCardScreenshotAdvanceRequest;
class GetAsyncJobResultRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            jobId: 'JobId',
        };
    }
    static types() {
        return {
            jobId: 'string',
        };
    }
}
exports.GetAsyncJobResultRequest = GetAsyncJobResultRequest;
class GetAsyncJobResultResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: GetAsyncJobResultResponseData,
        };
    }
}
exports.GetAsyncJobResultResponse = GetAsyncJobResultResponse;
class TrimDocumentRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            fileURL: 'FileURL',
            fileType: 'FileType',
            outputType: 'OutputType',
        };
    }
    static types() {
        return {
            fileURL: 'string',
            fileType: 'string',
            outputType: 'string',
        };
    }
}
exports.TrimDocumentRequest = TrimDocumentRequest;
class TrimDocumentResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: TrimDocumentResponseData,
        };
    }
}
exports.TrimDocumentResponse = TrimDocumentResponse;
class TrimDocumentAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            fileURLObject: 'FileURLObject',
            fileType: 'FileType',
            outputType: 'OutputType',
        };
    }
    static types() {
        return {
            fileURLObject: 'Readable',
            fileType: 'string',
            outputType: 'string',
        };
    }
}
exports.TrimDocumentAdvanceRequest = TrimDocumentAdvanceRequest;
class RecognizeChinapassportRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURL: 'ImageURL',
        };
    }
    static types() {
        return {
            imageURL: 'string',
        };
    }
}
exports.RecognizeChinapassportRequest = RecognizeChinapassportRequest;
class RecognizeChinapassportResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: RecognizeChinapassportResponseData,
        };
    }
}
exports.RecognizeChinapassportResponse = RecognizeChinapassportResponse;
class RecognizeChinapassportAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURLObject: 'ImageURLObject',
        };
    }
    static types() {
        return {
            imageURLObject: 'Readable',
        };
    }
}
exports.RecognizeChinapassportAdvanceRequest = RecognizeChinapassportAdvanceRequest;
class RecognizeVerificationcodeRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURL: 'ImageURL',
        };
    }
    static types() {
        return {
            imageURL: 'string',
        };
    }
}
exports.RecognizeVerificationcodeRequest = RecognizeVerificationcodeRequest;
class RecognizeVerificationcodeResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: RecognizeVerificationcodeResponseData,
        };
    }
}
exports.RecognizeVerificationcodeResponse = RecognizeVerificationcodeResponse;
class RecognizeVerificationcodeAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURLObject: 'ImageURLObject',
        };
    }
    static types() {
        return {
            imageURLObject: 'Readable',
        };
    }
}
exports.RecognizeVerificationcodeAdvanceRequest = RecognizeVerificationcodeAdvanceRequest;
class RecognizePassportMRZRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURL: 'ImageURL',
        };
    }
    static types() {
        return {
            imageURL: 'string',
        };
    }
}
exports.RecognizePassportMRZRequest = RecognizePassportMRZRequest;
class RecognizePassportMRZResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: RecognizePassportMRZResponseData,
        };
    }
}
exports.RecognizePassportMRZResponse = RecognizePassportMRZResponse;
class RecognizePassportMRZAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURLObject: 'ImageURLObject',
        };
    }
    static types() {
        return {
            imageURLObject: 'Readable',
        };
    }
}
exports.RecognizePassportMRZAdvanceRequest = RecognizePassportMRZAdvanceRequest;
class RecognizeTakeoutOrderRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURL: 'ImageURL',
        };
    }
    static types() {
        return {
            imageURL: 'string',
        };
    }
}
exports.RecognizeTakeoutOrderRequest = RecognizeTakeoutOrderRequest;
class RecognizeTakeoutOrderResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: RecognizeTakeoutOrderResponseData,
        };
    }
}
exports.RecognizeTakeoutOrderResponse = RecognizeTakeoutOrderResponse;
class RecognizeTakeoutOrderAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURLObject: 'ImageURLObject',
        };
    }
    static types() {
        return {
            imageURLObject: 'Readable',
        };
    }
}
exports.RecognizeTakeoutOrderAdvanceRequest = RecognizeTakeoutOrderAdvanceRequest;
class RecognizeQrCodeRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            tasks: 'Tasks',
        };
    }
    static types() {
        return {
            tasks: { 'type': 'array', 'itemType': RecognizeQrCodeRequestTasks },
        };
    }
}
exports.RecognizeQrCodeRequest = RecognizeQrCodeRequest;
class RecognizeQrCodeResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: RecognizeQrCodeResponseData,
        };
    }
}
exports.RecognizeQrCodeResponse = RecognizeQrCodeResponse;
class RecognizeVATInvoiceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            fileURL: 'FileURL',
            fileType: 'FileType',
        };
    }
    static types() {
        return {
            fileURL: 'string',
            fileType: 'string',
        };
    }
}
exports.RecognizeVATInvoiceRequest = RecognizeVATInvoiceRequest;
class RecognizeVATInvoiceResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: RecognizeVATInvoiceResponseData,
        };
    }
}
exports.RecognizeVATInvoiceResponse = RecognizeVATInvoiceResponse;
class RecognizeVATInvoiceAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            fileURLObject: 'FileURLObject',
            fileType: 'FileType',
        };
    }
    static types() {
        return {
            fileURLObject: 'Readable',
            fileType: 'string',
        };
    }
}
exports.RecognizeVATInvoiceAdvanceRequest = RecognizeVATInvoiceAdvanceRequest;
class RecognizeCharacterRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURL: 'ImageURL',
            minHeight: 'MinHeight',
            outputProbability: 'OutputProbability',
        };
    }
    static types() {
        return {
            imageURL: 'string',
            minHeight: 'number',
            outputProbability: 'boolean',
        };
    }
}
exports.RecognizeCharacterRequest = RecognizeCharacterRequest;
class RecognizeCharacterResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: RecognizeCharacterResponseData,
        };
    }
}
exports.RecognizeCharacterResponse = RecognizeCharacterResponse;
class RecognizeCharacterAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURLObject: 'ImageURLObject',
            minHeight: 'MinHeight',
            outputProbability: 'OutputProbability',
        };
    }
    static types() {
        return {
            imageURLObject: 'Readable',
            minHeight: 'number',
            outputProbability: 'boolean',
        };
    }
}
exports.RecognizeCharacterAdvanceRequest = RecognizeCharacterAdvanceRequest;
class RecognizeTaxiInvoiceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURL: 'ImageURL',
        };
    }
    static types() {
        return {
            imageURL: 'string',
        };
    }
}
exports.RecognizeTaxiInvoiceRequest = RecognizeTaxiInvoiceRequest;
class RecognizeTaxiInvoiceResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: RecognizeTaxiInvoiceResponseData,
        };
    }
}
exports.RecognizeTaxiInvoiceResponse = RecognizeTaxiInvoiceResponse;
class RecognizeTaxiInvoiceAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURLObject: 'ImageURLObject',
        };
    }
    static types() {
        return {
            imageURLObject: 'Readable',
        };
    }
}
exports.RecognizeTaxiInvoiceAdvanceRequest = RecognizeTaxiInvoiceAdvanceRequest;
class RecognizeIdentityCardRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURL: 'ImageURL',
            side: 'Side',
        };
    }
    static types() {
        return {
            imageURL: 'string',
            side: 'string',
        };
    }
}
exports.RecognizeIdentityCardRequest = RecognizeIdentityCardRequest;
class RecognizeIdentityCardResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: RecognizeIdentityCardResponseData,
        };
    }
}
exports.RecognizeIdentityCardResponse = RecognizeIdentityCardResponse;
class RecognizeIdentityCardAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURLObject: 'ImageURLObject',
            side: 'Side',
        };
    }
    static types() {
        return {
            imageURLObject: 'Readable',
            side: 'string',
        };
    }
}
exports.RecognizeIdentityCardAdvanceRequest = RecognizeIdentityCardAdvanceRequest;
class RecognizeLicensePlateRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURL: 'ImageURL',
        };
    }
    static types() {
        return {
            imageURL: 'string',
        };
    }
}
exports.RecognizeLicensePlateRequest = RecognizeLicensePlateRequest;
class RecognizeLicensePlateResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: RecognizeLicensePlateResponseData,
        };
    }
}
exports.RecognizeLicensePlateResponse = RecognizeLicensePlateResponse;
class RecognizeLicensePlateAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURLObject: 'ImageURLObject',
        };
    }
    static types() {
        return {
            imageURLObject: 'Readable',
        };
    }
}
exports.RecognizeLicensePlateAdvanceRequest = RecognizeLicensePlateAdvanceRequest;
class RecognizeTableRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURL: 'ImageURL',
            outputFormat: 'OutputFormat',
            useFinanceModel: 'UseFinanceModel',
            assureDirection: 'AssureDirection',
            hasLine: 'HasLine',
            skipDetection: 'SkipDetection',
        };
    }
    static types() {
        return {
            imageURL: 'string',
            outputFormat: 'string',
            useFinanceModel: 'boolean',
            assureDirection: 'boolean',
            hasLine: 'boolean',
            skipDetection: 'boolean',
        };
    }
}
exports.RecognizeTableRequest = RecognizeTableRequest;
class RecognizeTableResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: RecognizeTableResponseData,
        };
    }
}
exports.RecognizeTableResponse = RecognizeTableResponse;
class RecognizeTableAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURLObject: 'ImageURLObject',
            outputFormat: 'OutputFormat',
            useFinanceModel: 'UseFinanceModel',
            assureDirection: 'AssureDirection',
            hasLine: 'HasLine',
            skipDetection: 'SkipDetection',
        };
    }
    static types() {
        return {
            imageURLObject: 'Readable',
            outputFormat: 'string',
            useFinanceModel: 'boolean',
            assureDirection: 'boolean',
            hasLine: 'boolean',
            skipDetection: 'boolean',
        };
    }
}
exports.RecognizeTableAdvanceRequest = RecognizeTableAdvanceRequest;
class RecognizeDrivingLicenseRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURL: 'ImageURL',
            side: 'Side',
        };
    }
    static types() {
        return {
            imageURL: 'string',
            side: 'string',
        };
    }
}
exports.RecognizeDrivingLicenseRequest = RecognizeDrivingLicenseRequest;
class RecognizeDrivingLicenseResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: RecognizeDrivingLicenseResponseData,
        };
    }
}
exports.RecognizeDrivingLicenseResponse = RecognizeDrivingLicenseResponse;
class RecognizeDrivingLicenseAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURLObject: 'ImageURLObject',
            side: 'Side',
        };
    }
    static types() {
        return {
            imageURLObject: 'Readable',
            side: 'string',
        };
    }
}
exports.RecognizeDrivingLicenseAdvanceRequest = RecognizeDrivingLicenseAdvanceRequest;
class RecognizeBankCardRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURL: 'ImageURL',
        };
    }
    static types() {
        return {
            imageURL: 'string',
        };
    }
}
exports.RecognizeBankCardRequest = RecognizeBankCardRequest;
class RecognizeBankCardResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: RecognizeBankCardResponseData,
        };
    }
}
exports.RecognizeBankCardResponse = RecognizeBankCardResponse;
class RecognizeBankCardAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURLObject: 'ImageURLObject',
        };
    }
    static types() {
        return {
            imageURLObject: 'Readable',
        };
    }
}
exports.RecognizeBankCardAdvanceRequest = RecognizeBankCardAdvanceRequest;
class RecognizeTrainTicketRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURL: 'ImageURL',
        };
    }
    static types() {
        return {
            imageURL: 'string',
        };
    }
}
exports.RecognizeTrainTicketRequest = RecognizeTrainTicketRequest;
class RecognizeTrainTicketResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: RecognizeTrainTicketResponseData,
        };
    }
}
exports.RecognizeTrainTicketResponse = RecognizeTrainTicketResponse;
class RecognizeTrainTicketAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURLObject: 'ImageURLObject',
        };
    }
    static types() {
        return {
            imageURLObject: 'Readable',
        };
    }
}
exports.RecognizeTrainTicketAdvanceRequest = RecognizeTrainTicketAdvanceRequest;
class RecognizeDriverLicenseRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURL: 'ImageURL',
            side: 'Side',
        };
    }
    static types() {
        return {
            imageURL: 'string',
            side: 'string',
        };
    }
}
exports.RecognizeDriverLicenseRequest = RecognizeDriverLicenseRequest;
class RecognizeDriverLicenseResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: RecognizeDriverLicenseResponseData,
        };
    }
}
exports.RecognizeDriverLicenseResponse = RecognizeDriverLicenseResponse;
class RecognizeDriverLicenseAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURLObject: 'ImageURLObject',
            side: 'Side',
        };
    }
    static types() {
        return {
            imageURLObject: 'Readable',
            side: 'string',
        };
    }
}
exports.RecognizeDriverLicenseAdvanceRequest = RecognizeDriverLicenseAdvanceRequest;
class RecognizeAccountPageRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURL: 'ImageURL',
        };
    }
    static types() {
        return {
            imageURL: 'string',
        };
    }
}
exports.RecognizeAccountPageRequest = RecognizeAccountPageRequest;
class RecognizeAccountPageResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: RecognizeAccountPageResponseData,
        };
    }
}
exports.RecognizeAccountPageResponse = RecognizeAccountPageResponse;
class RecognizeAccountPageAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURLObject: 'ImageURLObject',
        };
    }
    static types() {
        return {
            imageURLObject: 'Readable',
        };
    }
}
exports.RecognizeAccountPageAdvanceRequest = RecognizeAccountPageAdvanceRequest;
class RecognizeStampRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURL: 'ImageURL',
        };
    }
    static types() {
        return {
            imageURL: 'string',
        };
    }
}
exports.RecognizeStampRequest = RecognizeStampRequest;
class RecognizeStampResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: RecognizeStampResponseData,
        };
    }
}
exports.RecognizeStampResponse = RecognizeStampResponse;
class RecognizeStampAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURLObject: 'ImageURLObject',
        };
    }
    static types() {
        return {
            imageURLObject: 'Readable',
        };
    }
}
exports.RecognizeStampAdvanceRequest = RecognizeStampAdvanceRequest;
class RecognizeBusinessCardRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURL: 'ImageURL',
        };
    }
    static types() {
        return {
            imageURL: 'string',
        };
    }
}
exports.RecognizeBusinessCardRequest = RecognizeBusinessCardRequest;
class RecognizeBusinessCardResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: RecognizeBusinessCardResponseData,
        };
    }
}
exports.RecognizeBusinessCardResponse = RecognizeBusinessCardResponse;
class RecognizeBusinessCardAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURLObject: 'ImageURLObject',
        };
    }
    static types() {
        return {
            imageURLObject: 'Readable',
        };
    }
}
exports.RecognizeBusinessCardAdvanceRequest = RecognizeBusinessCardAdvanceRequest;
class RecognizeVINCodeRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURL: 'ImageURL',
        };
    }
    static types() {
        return {
            imageURL: 'string',
        };
    }
}
exports.RecognizeVINCodeRequest = RecognizeVINCodeRequest;
class RecognizeVINCodeResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: RecognizeVINCodeResponseData,
        };
    }
}
exports.RecognizeVINCodeResponse = RecognizeVINCodeResponse;
class RecognizeVINCodeAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURLObject: 'ImageURLObject',
        };
    }
    static types() {
        return {
            imageURLObject: 'Readable',
        };
    }
}
exports.RecognizeVINCodeAdvanceRequest = RecognizeVINCodeAdvanceRequest;
class RecognizeBusinessLicenseRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURL: 'ImageURL',
        };
    }
    static types() {
        return {
            imageURL: 'string',
        };
    }
}
exports.RecognizeBusinessLicenseRequest = RecognizeBusinessLicenseRequest;
class RecognizeBusinessLicenseResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'RequestId',
            data: 'Data',
        };
    }
    static types() {
        return {
            requestId: 'string',
            data: RecognizeBusinessLicenseResponseData,
        };
    }
}
exports.RecognizeBusinessLicenseResponse = RecognizeBusinessLicenseResponse;
class RecognizeBusinessLicenseAdvanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURLObject: 'ImageURLObject',
        };
    }
    static types() {
        return {
            imageURLObject: 'Readable',
        };
    }
}
exports.RecognizeBusinessLicenseAdvanceRequest = RecognizeBusinessLicenseAdvanceRequest;
class RecognizePoiNameResponseDataSignboardsTexts extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            label: 'Label',
            score: 'Score',
            tag: 'Tag',
            type: 'Type',
            points: 'Points',
        };
    }
    static types() {
        return {
            label: 'string',
            score: 'number',
            tag: 'string',
            type: 'string',
            points: { 'type': 'array', 'itemType': 'number' },
        };
    }
}
exports.RecognizePoiNameResponseDataSignboardsTexts = RecognizePoiNameResponseDataSignboardsTexts;
class RecognizePoiNameResponseDataSignboards extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            texts: 'Texts',
        };
    }
    static types() {
        return {
            texts: { 'type': 'array', 'itemType': RecognizePoiNameResponseDataSignboardsTexts },
        };
    }
}
exports.RecognizePoiNameResponseDataSignboards = RecognizePoiNameResponseDataSignboards;
class RecognizePoiNameResponseDataSummary extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            brand: 'Brand',
            score: 'Score',
        };
    }
    static types() {
        return {
            brand: 'string',
            score: 'number',
        };
    }
}
exports.RecognizePoiNameResponseDataSummary = RecognizePoiNameResponseDataSummary;
class RecognizePoiNameResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            signboards: 'Signboards',
            summary: 'Summary',
        };
    }
    static types() {
        return {
            signboards: { 'type': 'array', 'itemType': RecognizePoiNameResponseDataSignboards },
            summary: RecognizePoiNameResponseDataSummary,
        };
    }
}
exports.RecognizePoiNameResponseData = RecognizePoiNameResponseData;
class DetectCardScreenshotResponseDataSpoofResultResultMap extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            screenScore: 'ScreenScore',
            screenThreshold: 'ScreenThreshold',
        };
    }
    static types() {
        return {
            screenScore: 'number',
            screenThreshold: 'number',
        };
    }
}
exports.DetectCardScreenshotResponseDataSpoofResultResultMap = DetectCardScreenshotResponseDataSpoofResultResultMap;
class DetectCardScreenshotResponseDataSpoofResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            isSpoof: 'IsSpoof',
            resultMap: 'ResultMap',
        };
    }
    static types() {
        return {
            isSpoof: 'boolean',
            resultMap: DetectCardScreenshotResponseDataSpoofResultResultMap,
        };
    }
}
exports.DetectCardScreenshotResponseDataSpoofResult = DetectCardScreenshotResponseDataSpoofResult;
class DetectCardScreenshotResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            isCard: 'IsCard',
            isBlur: 'IsBlur',
            spoofResult: 'SpoofResult',
        };
    }
    static types() {
        return {
            isCard: 'boolean',
            isBlur: 'boolean',
            spoofResult: DetectCardScreenshotResponseDataSpoofResult,
        };
    }
}
exports.DetectCardScreenshotResponseData = DetectCardScreenshotResponseData;
class GetAsyncJobResultResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            errorCode: 'ErrorCode',
            errorMessage: 'ErrorMessage',
            jobId: 'JobId',
            result: 'Result',
            status: 'Status',
        };
    }
    static types() {
        return {
            errorCode: 'string',
            errorMessage: 'string',
            jobId: 'string',
            result: 'string',
            status: 'string',
        };
    }
}
exports.GetAsyncJobResultResponseData = GetAsyncJobResultResponseData;
class TrimDocumentResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            content: 'Content',
        };
    }
    static types() {
        return {
            content: 'string',
        };
    }
}
exports.TrimDocumentResponseData = TrimDocumentResponseData;
class RecognizeChinapassportResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            authority: 'Authority',
            birthDate: 'BirthDate',
            birthDay: 'BirthDay',
            birthPlace: 'BirthPlace',
            birthPlaceRaw: 'BirthPlaceRaw',
            country: 'Country',
            expiryDate: 'ExpiryDate',
            expiryDay: 'ExpiryDay',
            issueDate: 'IssueDate',
            issuePlace: 'IssuePlace',
            issuePlaceRaw: 'IssuePlaceRaw',
            lineZero: 'LineZero',
            lineOne: 'LineOne',
            name: 'Name',
            nameChinese: 'NameChinese',
            nameChineseRaw: 'NameChineseRaw',
            passportNo: 'PassportNo',
            personId: 'PersonId',
            sex: 'Sex',
            sourceCountry: 'SourceCountry',
            success: 'Success',
            type: 'Type',
        };
    }
    static types() {
        return {
            authority: 'string',
            birthDate: 'string',
            birthDay: 'string',
            birthPlace: 'string',
            birthPlaceRaw: 'string',
            country: 'string',
            expiryDate: 'string',
            expiryDay: 'string',
            issueDate: 'string',
            issuePlace: 'string',
            issuePlaceRaw: 'string',
            lineZero: 'string',
            lineOne: 'string',
            name: 'string',
            nameChinese: 'string',
            nameChineseRaw: 'string',
            passportNo: 'string',
            personId: 'string',
            sex: 'string',
            sourceCountry: 'string',
            success: 'boolean',
            type: 'string',
        };
    }
}
exports.RecognizeChinapassportResponseData = RecognizeChinapassportResponseData;
class RecognizeVerificationcodeResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            content: 'Content',
        };
    }
    static types() {
        return {
            content: 'string',
        };
    }
}
exports.RecognizeVerificationcodeResponseData = RecognizeVerificationcodeResponseData;
class RecognizePassportMRZResponseDataRegions extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            name: 'Name',
            recognitionScore: 'RecognitionScore',
            content: 'Content',
            detectionScore: 'DetectionScore',
            bandBoxes: 'BandBoxes',
        };
    }
    static types() {
        return {
            name: 'string',
            recognitionScore: 'number',
            content: 'string',
            detectionScore: 'number',
            bandBoxes: { 'type': 'array', 'itemType': 'number' },
        };
    }
}
exports.RecognizePassportMRZResponseDataRegions = RecognizePassportMRZResponseDataRegions;
class RecognizePassportMRZResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            regions: 'Regions',
        };
    }
    static types() {
        return {
            regions: { 'type': 'array', 'itemType': RecognizePassportMRZResponseDataRegions },
        };
    }
}
exports.RecognizePassportMRZResponseData = RecognizePassportMRZResponseData;
class RecognizeTakeoutOrderResponseDataElements extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            score: 'Score',
            name: 'Name',
            value: 'Value',
            boxes: 'Boxes',
        };
    }
    static types() {
        return {
            score: 'number',
            name: 'string',
            value: 'string',
            boxes: { 'type': 'array', 'itemType': 'number' },
        };
    }
}
exports.RecognizeTakeoutOrderResponseDataElements = RecognizeTakeoutOrderResponseDataElements;
class RecognizeTakeoutOrderResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            elements: 'Elements',
        };
    }
    static types() {
        return {
            elements: { 'type': 'array', 'itemType': RecognizeTakeoutOrderResponseDataElements },
        };
    }
}
exports.RecognizeTakeoutOrderResponseData = RecognizeTakeoutOrderResponseData;
class RecognizeQrCodeRequestTasks extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            imageURL: 'ImageURL',
        };
    }
    static types() {
        return {
            imageURL: 'string',
        };
    }
}
exports.RecognizeQrCodeRequestTasks = RecognizeQrCodeRequestTasks;
class RecognizeQrCodeResponseDataElementsResults extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            label: 'Label',
            suggestion: 'Suggestion',
            rate: 'Rate',
            qrCodesData: 'QrCodesData',
        };
    }
    static types() {
        return {
            label: 'string',
            suggestion: 'string',
            rate: 'number',
            qrCodesData: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.RecognizeQrCodeResponseDataElementsResults = RecognizeQrCodeResponseDataElementsResults;
class RecognizeQrCodeResponseDataElements extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            taskId: 'TaskId',
            imageURL: 'ImageURL',
            results: 'Results',
        };
    }
    static types() {
        return {
            taskId: 'string',
            imageURL: 'string',
            results: { 'type': 'array', 'itemType': RecognizeQrCodeResponseDataElementsResults },
        };
    }
}
exports.RecognizeQrCodeResponseDataElements = RecognizeQrCodeResponseDataElements;
class RecognizeQrCodeResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            elements: 'Elements',
        };
    }
    static types() {
        return {
            elements: { 'type': 'array', 'itemType': RecognizeQrCodeResponseDataElements },
        };
    }
}
exports.RecognizeQrCodeResponseData = RecognizeQrCodeResponseData;
class RecognizeVATInvoiceResponseDataContent extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            invoiceCode: 'InvoiceCode',
            invoiceNo: 'InvoiceNo',
            invoiceDate: 'InvoiceDate',
            antiFakeCode: 'AntiFakeCode',
            payerName: 'PayerName',
            payerRegisterNo: 'PayerRegisterNo',
            payerAddress: 'PayerAddress',
            payerBankName: 'PayerBankName',
            withoutTaxAmount: 'WithoutTaxAmount',
            taxAmount: 'TaxAmount',
            sumAmount: 'SumAmount',
            invoiceAmount: 'InvoiceAmount',
            payeeName: 'PayeeName',
            payeeRegisterNo: 'PayeeRegisterNo',
            payeeAddress: 'PayeeAddress',
            payeeBankName: 'PayeeBankName',
            payee: 'Payee',
            checker: 'Checker',
            clerk: 'Clerk',
        };
    }
    static types() {
        return {
            invoiceCode: 'string',
            invoiceNo: 'string',
            invoiceDate: 'string',
            antiFakeCode: 'string',
            payerName: 'string',
            payerRegisterNo: 'string',
            payerAddress: 'string',
            payerBankName: 'string',
            withoutTaxAmount: 'string',
            taxAmount: 'string',
            sumAmount: 'string',
            invoiceAmount: 'string',
            payeeName: 'string',
            payeeRegisterNo: 'string',
            payeeAddress: 'string',
            payeeBankName: 'string',
            payee: 'string',
            checker: 'string',
            clerk: 'string',
        };
    }
}
exports.RecognizeVATInvoiceResponseDataContent = RecognizeVATInvoiceResponseDataContent;
class RecognizeVATInvoiceResponseDataBox extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            invoiceCodes: 'InvoiceCodes',
            invoiceNoes: 'InvoiceNoes',
            invoiceDates: 'InvoiceDates',
            invoiceFakeCodes: 'InvoiceFakeCodes',
            payerNames: 'PayerNames',
            payerRegisterNoes: 'PayerRegisterNoes',
            payerAddresses: 'PayerAddresses',
            payerBankNames: 'PayerBankNames',
            withoutTaxAmounts: 'WithoutTaxAmounts',
            taxAmounts: 'TaxAmounts',
            sumAmounts: 'SumAmounts',
            invoiceAmounts: 'InvoiceAmounts',
            payeeNames: 'PayeeNames',
            payeeRegisterNoes: 'PayeeRegisterNoes',
            payeeAddresses: 'PayeeAddresses',
            payeeBankNames: 'PayeeBankNames',
            payees: 'Payees',
            checkers: 'Checkers',
            clerks: 'Clerks',
        };
    }
    static types() {
        return {
            invoiceCodes: { 'type': 'array', 'itemType': 'number' },
            invoiceNoes: { 'type': 'array', 'itemType': 'number' },
            invoiceDates: { 'type': 'array', 'itemType': 'number' },
            invoiceFakeCodes: { 'type': 'array', 'itemType': 'number' },
            payerNames: { 'type': 'array', 'itemType': 'number' },
            payerRegisterNoes: { 'type': 'array', 'itemType': 'number' },
            payerAddresses: { 'type': 'array', 'itemType': 'number' },
            payerBankNames: { 'type': 'array', 'itemType': 'number' },
            withoutTaxAmounts: { 'type': 'array', 'itemType': 'number' },
            taxAmounts: { 'type': 'array', 'itemType': 'number' },
            sumAmounts: { 'type': 'array', 'itemType': 'number' },
            invoiceAmounts: { 'type': 'array', 'itemType': 'number' },
            payeeNames: { 'type': 'array', 'itemType': 'number' },
            payeeRegisterNoes: { 'type': 'array', 'itemType': 'number' },
            payeeAddresses: { 'type': 'array', 'itemType': 'number' },
            payeeBankNames: { 'type': 'array', 'itemType': 'number' },
            payees: { 'type': 'array', 'itemType': 'number' },
            checkers: { 'type': 'array', 'itemType': 'number' },
            clerks: { 'type': 'array', 'itemType': 'number' },
        };
    }
}
exports.RecognizeVATInvoiceResponseDataBox = RecognizeVATInvoiceResponseDataBox;
class RecognizeVATInvoiceResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            content: 'Content',
            box: 'Box',
        };
    }
    static types() {
        return {
            content: RecognizeVATInvoiceResponseDataContent,
            box: RecognizeVATInvoiceResponseDataBox,
        };
    }
}
exports.RecognizeVATInvoiceResponseData = RecognizeVATInvoiceResponseData;
class RecognizeCharacterResponseDataResultsTextRectangles extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            angle: 'Angle',
            left: 'Left',
            top: 'Top',
            width: 'Width',
            height: 'Height',
        };
    }
    static types() {
        return {
            angle: 'number',
            left: 'number',
            top: 'number',
            width: 'number',
            height: 'number',
        };
    }
}
exports.RecognizeCharacterResponseDataResultsTextRectangles = RecognizeCharacterResponseDataResultsTextRectangles;
class RecognizeCharacterResponseDataResults extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            probability: 'Probability',
            text: 'Text',
            textRectangles: 'TextRectangles',
        };
    }
    static types() {
        return {
            probability: 'number',
            text: 'string',
            textRectangles: RecognizeCharacterResponseDataResultsTextRectangles,
        };
    }
}
exports.RecognizeCharacterResponseDataResults = RecognizeCharacterResponseDataResults;
class RecognizeCharacterResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            results: 'Results',
        };
    }
    static types() {
        return {
            results: { 'type': 'array', 'itemType': RecognizeCharacterResponseDataResults },
        };
    }
}
exports.RecognizeCharacterResponseData = RecognizeCharacterResponseData;
class RecognizeTaxiInvoiceResponseDataInvoicesItemsItemRoiCenter extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            x: 'X',
            y: 'Y',
        };
    }
    static types() {
        return {
            x: 'number',
            y: 'number',
        };
    }
}
exports.RecognizeTaxiInvoiceResponseDataInvoicesItemsItemRoiCenter = RecognizeTaxiInvoiceResponseDataInvoicesItemsItemRoiCenter;
class RecognizeTaxiInvoiceResponseDataInvoicesItemsItemRoiSize extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            h: 'H',
            w: 'W',
        };
    }
    static types() {
        return {
            h: 'number',
            w: 'number',
        };
    }
}
exports.RecognizeTaxiInvoiceResponseDataInvoicesItemsItemRoiSize = RecognizeTaxiInvoiceResponseDataInvoicesItemsItemRoiSize;
class RecognizeTaxiInvoiceResponseDataInvoicesItemsItemRoi extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            angle: 'Angle',
            center: 'Center',
            size: 'Size',
        };
    }
    static types() {
        return {
            angle: 'number',
            center: RecognizeTaxiInvoiceResponseDataInvoicesItemsItemRoiCenter,
            size: RecognizeTaxiInvoiceResponseDataInvoicesItemsItemRoiSize,
        };
    }
}
exports.RecognizeTaxiInvoiceResponseDataInvoicesItemsItemRoi = RecognizeTaxiInvoiceResponseDataInvoicesItemsItemRoi;
class RecognizeTaxiInvoiceResponseDataInvoicesItems extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            text: 'Text',
            itemRoi: 'ItemRoi',
        };
    }
    static types() {
        return {
            text: 'string',
            itemRoi: RecognizeTaxiInvoiceResponseDataInvoicesItemsItemRoi,
        };
    }
}
exports.RecognizeTaxiInvoiceResponseDataInvoicesItems = RecognizeTaxiInvoiceResponseDataInvoicesItems;
class RecognizeTaxiInvoiceResponseDataInvoicesInvoiceRoi extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            h: 'H',
            w: 'W',
            x: 'X',
            y: 'Y',
        };
    }
    static types() {
        return {
            h: 'number',
            w: 'number',
            x: 'number',
            y: 'number',
        };
    }
}
exports.RecognizeTaxiInvoiceResponseDataInvoicesInvoiceRoi = RecognizeTaxiInvoiceResponseDataInvoicesInvoiceRoi;
class RecognizeTaxiInvoiceResponseDataInvoices extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            rotateType: 'RotateType',
            items: 'Items',
            invoiceRoi: 'InvoiceRoi',
        };
    }
    static types() {
        return {
            rotateType: 'number',
            items: { 'type': 'array', 'itemType': RecognizeTaxiInvoiceResponseDataInvoicesItems },
            invoiceRoi: RecognizeTaxiInvoiceResponseDataInvoicesInvoiceRoi,
        };
    }
}
exports.RecognizeTaxiInvoiceResponseDataInvoices = RecognizeTaxiInvoiceResponseDataInvoices;
class RecognizeTaxiInvoiceResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            invoices: 'Invoices',
        };
    }
    static types() {
        return {
            invoices: { 'type': 'array', 'itemType': RecognizeTaxiInvoiceResponseDataInvoices },
        };
    }
}
exports.RecognizeTaxiInvoiceResponseData = RecognizeTaxiInvoiceResponseData;
class RecognizeIdentityCardResponseDataFrontResultCardAreas extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            x: 'X',
            y: 'Y',
        };
    }
    static types() {
        return {
            x: 'number',
            y: 'number',
        };
    }
}
exports.RecognizeIdentityCardResponseDataFrontResultCardAreas = RecognizeIdentityCardResponseDataFrontResultCardAreas;
class RecognizeIdentityCardResponseDataFrontResultFaceRectVertices extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            x: 'X',
            y: 'Y',
        };
    }
    static types() {
        return {
            x: 'number',
            y: 'number',
        };
    }
}
exports.RecognizeIdentityCardResponseDataFrontResultFaceRectVertices = RecognizeIdentityCardResponseDataFrontResultFaceRectVertices;
class RecognizeIdentityCardResponseDataFrontResultFaceRectangleCenter extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            x: 'X',
            y: 'Y',
        };
    }
    static types() {
        return {
            x: 'number',
            y: 'number',
        };
    }
}
exports.RecognizeIdentityCardResponseDataFrontResultFaceRectangleCenter = RecognizeIdentityCardResponseDataFrontResultFaceRectangleCenter;
class RecognizeIdentityCardResponseDataFrontResultFaceRectangleSize extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            height: 'Height',
            width: 'Width',
        };
    }
    static types() {
        return {
            height: 'number',
            width: 'number',
        };
    }
}
exports.RecognizeIdentityCardResponseDataFrontResultFaceRectangleSize = RecognizeIdentityCardResponseDataFrontResultFaceRectangleSize;
class RecognizeIdentityCardResponseDataFrontResultFaceRectangle extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            angle: 'Angle',
            center: 'Center',
            size: 'Size',
        };
    }
    static types() {
        return {
            angle: 'number',
            center: RecognizeIdentityCardResponseDataFrontResultFaceRectangleCenter,
            size: RecognizeIdentityCardResponseDataFrontResultFaceRectangleSize,
        };
    }
}
exports.RecognizeIdentityCardResponseDataFrontResultFaceRectangle = RecognizeIdentityCardResponseDataFrontResultFaceRectangle;
class RecognizeIdentityCardResponseDataFrontResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            address: 'Address',
            name: 'Name',
            nationality: 'Nationality',
            IDNumber: 'IDNumber',
            gender: 'Gender',
            birthDate: 'BirthDate',
            cardAreas: 'CardAreas',
            faceRectVertices: 'FaceRectVertices',
            faceRectangle: 'FaceRectangle',
        };
    }
    static types() {
        return {
            address: 'string',
            name: 'string',
            nationality: 'string',
            IDNumber: 'string',
            gender: 'string',
            birthDate: 'string',
            cardAreas: { 'type': 'array', 'itemType': RecognizeIdentityCardResponseDataFrontResultCardAreas },
            faceRectVertices: { 'type': 'array', 'itemType': RecognizeIdentityCardResponseDataFrontResultFaceRectVertices },
            faceRectangle: RecognizeIdentityCardResponseDataFrontResultFaceRectangle,
        };
    }
}
exports.RecognizeIdentityCardResponseDataFrontResult = RecognizeIdentityCardResponseDataFrontResult;
class RecognizeIdentityCardResponseDataBackResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            startDate: 'StartDate',
            endDate: 'EndDate',
            issue: 'Issue',
        };
    }
    static types() {
        return {
            startDate: 'string',
            endDate: 'string',
            issue: 'string',
        };
    }
}
exports.RecognizeIdentityCardResponseDataBackResult = RecognizeIdentityCardResponseDataBackResult;
class RecognizeIdentityCardResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            frontResult: 'FrontResult',
            backResult: 'BackResult',
        };
    }
    static types() {
        return {
            frontResult: RecognizeIdentityCardResponseDataFrontResult,
            backResult: RecognizeIdentityCardResponseDataBackResult,
        };
    }
}
exports.RecognizeIdentityCardResponseData = RecognizeIdentityCardResponseData;
class RecognizeLicensePlateResponseDataPlatesRoi extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            h: 'H',
            w: 'W',
            x: 'X',
            y: 'Y',
        };
    }
    static types() {
        return {
            h: 'number',
            w: 'number',
            x: 'number',
            y: 'number',
        };
    }
}
exports.RecognizeLicensePlateResponseDataPlatesRoi = RecognizeLicensePlateResponseDataPlatesRoi;
class RecognizeLicensePlateResponseDataPlates extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            confidence: 'Confidence',
            plateNumber: 'PlateNumber',
            plateType: 'PlateType',
            plateTypeConfidence: 'PlateTypeConfidence',
            roi: 'Roi',
        };
    }
    static types() {
        return {
            confidence: 'number',
            plateNumber: 'string',
            plateType: 'string',
            plateTypeConfidence: 'number',
            roi: RecognizeLicensePlateResponseDataPlatesRoi,
        };
    }
}
exports.RecognizeLicensePlateResponseDataPlates = RecognizeLicensePlateResponseDataPlates;
class RecognizeLicensePlateResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            plates: 'Plates',
        };
    }
    static types() {
        return {
            plates: { 'type': 'array', 'itemType': RecognizeLicensePlateResponseDataPlates },
        };
    }
}
exports.RecognizeLicensePlateResponseData = RecognizeLicensePlateResponseData;
class RecognizeTableResponseDataTablesTableRowsTableColumns extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            startColumn: 'StartColumn',
            startRow: 'StartRow',
            endColumn: 'EndColumn',
            endRow: 'EndRow',
            height: 'Height',
            width: 'Width',
            texts: 'Texts',
        };
    }
    static types() {
        return {
            startColumn: 'number',
            startRow: 'number',
            endColumn: 'number',
            endRow: 'number',
            height: 'number',
            width: 'number',
            texts: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.RecognizeTableResponseDataTablesTableRowsTableColumns = RecognizeTableResponseDataTablesTableRowsTableColumns;
class RecognizeTableResponseDataTablesTableRows extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            tableColumns: 'TableColumns',
        };
    }
    static types() {
        return {
            tableColumns: { 'type': 'array', 'itemType': RecognizeTableResponseDataTablesTableRowsTableColumns },
        };
    }
}
exports.RecognizeTableResponseDataTablesTableRows = RecognizeTableResponseDataTablesTableRows;
class RecognizeTableResponseDataTables extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            tableRows: 'TableRows',
            head: 'Head',
            tail: 'Tail',
        };
    }
    static types() {
        return {
            tableRows: { 'type': 'array', 'itemType': RecognizeTableResponseDataTablesTableRows },
            head: { 'type': 'array', 'itemType': 'string' },
            tail: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.RecognizeTableResponseDataTables = RecognizeTableResponseDataTables;
class RecognizeTableResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            fileContent: 'FileContent',
            tables: 'Tables',
        };
    }
    static types() {
        return {
            fileContent: 'string',
            tables: { 'type': 'array', 'itemType': RecognizeTableResponseDataTables },
        };
    }
}
exports.RecognizeTableResponseData = RecognizeTableResponseData;
class RecognizeDrivingLicenseResponseDataFaceResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            plateNumber: 'PlateNumber',
            vehicleType: 'VehicleType',
            owner: 'Owner',
            useCharacter: 'UseCharacter',
            address: 'Address',
            model: 'Model',
            vin: 'Vin',
            engineNumber: 'EngineNumber',
            registerDate: 'RegisterDate',
            issueDate: 'IssueDate',
        };
    }
    static types() {
        return {
            plateNumber: 'string',
            vehicleType: 'string',
            owner: 'string',
            useCharacter: 'string',
            address: 'string',
            model: 'string',
            vin: 'string',
            engineNumber: 'string',
            registerDate: 'string',
            issueDate: 'string',
        };
    }
}
exports.RecognizeDrivingLicenseResponseDataFaceResult = RecognizeDrivingLicenseResponseDataFaceResult;
class RecognizeDrivingLicenseResponseDataBackResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            approvedPassengerCapacity: 'ApprovedPassengerCapacity',
            approvedLoad: 'ApprovedLoad',
            fileNumber: 'FileNumber',
            grossMass: 'GrossMass',
            energyType: 'EnergyType',
            inspectionRecord: 'InspectionRecord',
            overallDimension: 'OverallDimension',
            tractionMass: 'TractionMass',
            unladenMass: 'UnladenMass',
            plateNumber: 'PlateNumber',
        };
    }
    static types() {
        return {
            approvedPassengerCapacity: 'string',
            approvedLoad: 'string',
            fileNumber: 'string',
            grossMass: 'string',
            energyType: 'string',
            inspectionRecord: 'string',
            overallDimension: 'string',
            tractionMass: 'string',
            unladenMass: 'string',
            plateNumber: 'string',
        };
    }
}
exports.RecognizeDrivingLicenseResponseDataBackResult = RecognizeDrivingLicenseResponseDataBackResult;
class RecognizeDrivingLicenseResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            faceResult: 'FaceResult',
            backResult: 'BackResult',
        };
    }
    static types() {
        return {
            faceResult: RecognizeDrivingLicenseResponseDataFaceResult,
            backResult: RecognizeDrivingLicenseResponseDataBackResult,
        };
    }
}
exports.RecognizeDrivingLicenseResponseData = RecognizeDrivingLicenseResponseData;
class RecognizeBankCardResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            bankName: 'BankName',
            cardNumber: 'CardNumber',
            validDate: 'ValidDate',
        };
    }
    static types() {
        return {
            bankName: 'string',
            cardNumber: 'string',
            validDate: 'string',
        };
    }
}
exports.RecognizeBankCardResponseData = RecognizeBankCardResponseData;
class RecognizeTrainTicketResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            date: 'Date',
            destination: 'Destination',
            level: 'Level',
            number: 'Number',
            name: 'Name',
            departureStation: 'DepartureStation',
            seat: 'Seat',
            price: 'Price',
        };
    }
    static types() {
        return {
            date: 'string',
            destination: 'string',
            level: 'string',
            number: 'string',
            name: 'string',
            departureStation: 'string',
            seat: 'string',
            price: 'number',
        };
    }
}
exports.RecognizeTrainTicketResponseData = RecognizeTrainTicketResponseData;
class RecognizeDriverLicenseResponseDataFaceResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            name: 'Name',
            licenseNumber: 'LicenseNumber',
            vehicleType: 'VehicleType',
            startDate: 'StartDate',
            endDate: 'EndDate',
            issueDate: 'IssueDate',
            address: 'Address',
            gender: 'Gender',
        };
    }
    static types() {
        return {
            name: 'string',
            licenseNumber: 'string',
            vehicleType: 'string',
            startDate: 'string',
            endDate: 'string',
            issueDate: 'string',
            address: 'string',
            gender: 'string',
        };
    }
}
exports.RecognizeDriverLicenseResponseDataFaceResult = RecognizeDriverLicenseResponseDataFaceResult;
class RecognizeDriverLicenseResponseDataBackResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            archiveNumber: 'ArchiveNumber',
        };
    }
    static types() {
        return {
            archiveNumber: 'string',
        };
    }
}
exports.RecognizeDriverLicenseResponseDataBackResult = RecognizeDriverLicenseResponseDataBackResult;
class RecognizeDriverLicenseResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            faceResult: 'FaceResult',
            backResult: 'BackResult',
        };
    }
    static types() {
        return {
            faceResult: RecognizeDriverLicenseResponseDataFaceResult,
            backResult: RecognizeDriverLicenseResponseDataBackResult,
        };
    }
}
exports.RecognizeDriverLicenseResponseData = RecognizeDriverLicenseResponseData;
class RecognizeAccountPageResponseDataInvalidStampAreas extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            left: 'Left',
            top: 'Top',
            height: 'Height',
            width: 'Width',
        };
    }
    static types() {
        return {
            left: 'number',
            top: 'number',
            height: 'number',
            width: 'number',
        };
    }
}
exports.RecognizeAccountPageResponseDataInvalidStampAreas = RecognizeAccountPageResponseDataInvalidStampAreas;
class RecognizeAccountPageResponseDataUndertakeStampAreas extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            left: 'Left',
            top: 'Top',
            height: 'Height',
            width: 'Width',
        };
    }
    static types() {
        return {
            left: 'number',
            top: 'number',
            height: 'number',
            width: 'number',
        };
    }
}
exports.RecognizeAccountPageResponseDataUndertakeStampAreas = RecognizeAccountPageResponseDataUndertakeStampAreas;
class RecognizeAccountPageResponseDataRegisterStampAreas extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            left: 'Left',
            top: 'Top',
            height: 'Height',
            width: 'Width',
        };
    }
    static types() {
        return {
            left: 'number',
            top: 'number',
            height: 'number',
            width: 'number',
        };
    }
}
exports.RecognizeAccountPageResponseDataRegisterStampAreas = RecognizeAccountPageResponseDataRegisterStampAreas;
class RecognizeAccountPageResponseDataOtherStampAreas extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            left: 'Left',
            top: 'Top',
            height: 'Height',
            width: 'Width',
        };
    }
    static types() {
        return {
            left: 'number',
            top: 'number',
            height: 'number',
            width: 'number',
        };
    }
}
exports.RecognizeAccountPageResponseDataOtherStampAreas = RecognizeAccountPageResponseDataOtherStampAreas;
class RecognizeAccountPageResponseDataTitleArea extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            left: 'Left',
            top: 'Top',
            height: 'Height',
            width: 'Width',
        };
    }
    static types() {
        return {
            left: 'number',
            top: 'number',
            height: 'number',
            width: 'number',
        };
    }
}
exports.RecognizeAccountPageResponseDataTitleArea = RecognizeAccountPageResponseDataTitleArea;
class RecognizeAccountPageResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            angle: 'Angle',
            name: 'Name',
            gender: 'Gender',
            relation: 'Relation',
            birthPlace: 'BirthPlace',
            nationality: 'Nationality',
            nativePlace: 'NativePlace',
            birthDate: 'BirthDate',
            IDNumber: 'IDNumber',
            invalidStampAreas: 'InvalidStampAreas',
            undertakeStampAreas: 'UndertakeStampAreas',
            registerStampAreas: 'RegisterStampAreas',
            otherStampAreas: 'OtherStampAreas',
            titleArea: 'TitleArea',
        };
    }
    static types() {
        return {
            angle: 'number',
            name: 'string',
            gender: 'string',
            relation: 'string',
            birthPlace: 'string',
            nationality: 'string',
            nativePlace: 'string',
            birthDate: 'string',
            IDNumber: 'string',
            invalidStampAreas: { 'type': 'array', 'itemType': RecognizeAccountPageResponseDataInvalidStampAreas },
            undertakeStampAreas: { 'type': 'array', 'itemType': RecognizeAccountPageResponseDataUndertakeStampAreas },
            registerStampAreas: { 'type': 'array', 'itemType': RecognizeAccountPageResponseDataRegisterStampAreas },
            otherStampAreas: { 'type': 'array', 'itemType': RecognizeAccountPageResponseDataOtherStampAreas },
            titleArea: RecognizeAccountPageResponseDataTitleArea,
        };
    }
}
exports.RecognizeAccountPageResponseData = RecognizeAccountPageResponseData;
class RecognizeStampResponseDataResultsGeneralText extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            content: 'Content',
            confidence: 'Confidence',
        };
    }
    static types() {
        return {
            content: 'string',
            confidence: 'number',
        };
    }
}
exports.RecognizeStampResponseDataResultsGeneralText = RecognizeStampResponseDataResultsGeneralText;
class RecognizeStampResponseDataResultsRoi extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            left: 'Left',
            top: 'Top',
            width: 'Width',
            height: 'Height',
        };
    }
    static types() {
        return {
            left: 'number',
            top: 'number',
            width: 'number',
            height: 'number',
        };
    }
}
exports.RecognizeStampResponseDataResultsRoi = RecognizeStampResponseDataResultsRoi;
class RecognizeStampResponseDataResultsText extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            content: 'Content',
            confidence: 'Confidence',
        };
    }
    static types() {
        return {
            content: 'string',
            confidence: 'number',
        };
    }
}
exports.RecognizeStampResponseDataResultsText = RecognizeStampResponseDataResultsText;
class RecognizeStampResponseDataResults extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            generalText: 'GeneralText',
            roi: 'Roi',
            text: 'Text',
        };
    }
    static types() {
        return {
            generalText: { 'type': 'array', 'itemType': RecognizeStampResponseDataResultsGeneralText },
            roi: RecognizeStampResponseDataResultsRoi,
            text: RecognizeStampResponseDataResultsText,
        };
    }
}
exports.RecognizeStampResponseDataResults = RecognizeStampResponseDataResults;
class RecognizeStampResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            results: 'Results',
        };
    }
    static types() {
        return {
            results: { 'type': 'array', 'itemType': RecognizeStampResponseDataResults },
        };
    }
}
exports.RecognizeStampResponseData = RecognizeStampResponseData;
class RecognizeBusinessCardResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            name: 'Name',
            companies: 'Companies',
            departments: 'Departments',
            titles: 'Titles',
            cellPhoneNumbers: 'CellPhoneNumbers',
            officePhoneNumbers: 'OfficePhoneNumbers',
            addresses: 'Addresses',
            emails: 'Emails',
        };
    }
    static types() {
        return {
            name: 'string',
            companies: { 'type': 'array', 'itemType': 'string' },
            departments: { 'type': 'array', 'itemType': 'string' },
            titles: { 'type': 'array', 'itemType': 'string' },
            cellPhoneNumbers: { 'type': 'array', 'itemType': 'string' },
            officePhoneNumbers: { 'type': 'array', 'itemType': 'string' },
            addresses: { 'type': 'array', 'itemType': 'string' },
            emails: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.RecognizeBusinessCardResponseData = RecognizeBusinessCardResponseData;
class RecognizeVINCodeResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            vinCode: 'VinCode',
        };
    }
    static types() {
        return {
            vinCode: 'string',
        };
    }
}
exports.RecognizeVINCodeResponseData = RecognizeVINCodeResponseData;
class RecognizeBusinessLicenseResponseDataEmblem extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            top: 'Top',
            left: 'Left',
            height: 'Height',
            width: 'Width',
        };
    }
    static types() {
        return {
            top: 'number',
            left: 'number',
            height: 'number',
            width: 'number',
        };
    }
}
exports.RecognizeBusinessLicenseResponseDataEmblem = RecognizeBusinessLicenseResponseDataEmblem;
class RecognizeBusinessLicenseResponseDataTitle extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            top: 'Top',
            left: 'Left',
            height: 'Height',
            width: 'Width',
        };
    }
    static types() {
        return {
            top: 'number',
            left: 'number',
            height: 'number',
            width: 'number',
        };
    }
}
exports.RecognizeBusinessLicenseResponseDataTitle = RecognizeBusinessLicenseResponseDataTitle;
class RecognizeBusinessLicenseResponseDataStamp extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            top: 'Top',
            left: 'Left',
            height: 'Height',
            width: 'Width',
        };
    }
    static types() {
        return {
            top: 'number',
            left: 'number',
            height: 'number',
            width: 'number',
        };
    }
}
exports.RecognizeBusinessLicenseResponseDataStamp = RecognizeBusinessLicenseResponseDataStamp;
class RecognizeBusinessLicenseResponseDataQRCode extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            top: 'Top',
            left: 'Left',
            height: 'Height',
            width: 'Width',
        };
    }
    static types() {
        return {
            top: 'number',
            left: 'number',
            height: 'number',
            width: 'number',
        };
    }
}
exports.RecognizeBusinessLicenseResponseDataQRCode = RecognizeBusinessLicenseResponseDataQRCode;
class RecognizeBusinessLicenseResponseData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            angle: 'Angle',
            registerNumber: 'RegisterNumber',
            name: 'Name',
            type: 'Type',
            legalPerson: 'LegalPerson',
            establishDate: 'EstablishDate',
            validPeriod: 'ValidPeriod',
            address: 'Address',
            capital: 'Capital',
            business: 'Business',
            emblem: 'Emblem',
            title: 'Title',
            stamp: 'Stamp',
            QRCode: 'QRCode',
        };
    }
    static types() {
        return {
            angle: 'number',
            registerNumber: 'string',
            name: 'string',
            type: 'string',
            legalPerson: 'string',
            establishDate: 'string',
            validPeriod: 'string',
            address: 'string',
            capital: 'string',
            business: 'string',
            emblem: RecognizeBusinessLicenseResponseDataEmblem,
            title: RecognizeBusinessLicenseResponseDataTitle,
            stamp: RecognizeBusinessLicenseResponseDataStamp,
            QRCode: RecognizeBusinessLicenseResponseDataQRCode,
        };
    }
}
exports.RecognizeBusinessLicenseResponseData = RecognizeBusinessLicenseResponseData;
class Client extends rpc_client_1.default {
    constructor(config) {
        super(config);
        this._endpointRule = "regional";
        this.checkConfig(config);
        this._endpoint = this.getEndpoint("ocr", this._regionId, this._endpointRule, this._network, this._suffix, this._endpointMap, this._endpoint);
    }
    async recognizePoiName(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("RecognizePoiName", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new RecognizePoiNameResponse({}));
    }
    async recognizePoiNameAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let recognizePoiNamereq = new RecognizePoiNameRequest({});
        rpc_util_1.default.convert(request, recognizePoiNamereq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.imageURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        recognizePoiNamereq.imageURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let recognizePoiNameResp = await this.recognizePoiName(recognizePoiNamereq, runtime);
        return recognizePoiNameResp;
    }
    async detectCardScreenshot(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("DetectCardScreenshot", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new DetectCardScreenshotResponse({}));
    }
    async detectCardScreenshotAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let detectCardScreenshotreq = new DetectCardScreenshotRequest({});
        rpc_util_1.default.convert(request, detectCardScreenshotreq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.imageURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        detectCardScreenshotreq.imageURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let detectCardScreenshotResp = await this.detectCardScreenshot(detectCardScreenshotreq, runtime);
        return detectCardScreenshotResp;
    }
    async getAsyncJobResult(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("GetAsyncJobResult", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new GetAsyncJobResultResponse({}));
    }
    async trimDocument(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("TrimDocument", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new TrimDocumentResponse({}));
    }
    async trimDocumentAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let trimDocumentreq = new TrimDocumentRequest({});
        rpc_util_1.default.convert(request, trimDocumentreq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.fileURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        trimDocumentreq.fileURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let trimDocumentResp = await this.trimDocument(trimDocumentreq, runtime);
        return trimDocumentResp;
    }
    async recognizeChinapassport(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("RecognizeChinapassport", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new RecognizeChinapassportResponse({}));
    }
    async recognizeChinapassportAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let recognizeChinapassportreq = new RecognizeChinapassportRequest({});
        rpc_util_1.default.convert(request, recognizeChinapassportreq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.imageURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        recognizeChinapassportreq.imageURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let recognizeChinapassportResp = await this.recognizeChinapassport(recognizeChinapassportreq, runtime);
        return recognizeChinapassportResp;
    }
    async recognizeVerificationcode(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("RecognizeVerificationcode", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new RecognizeVerificationcodeResponse({}));
    }
    async recognizeVerificationcodeAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let recognizeVerificationcodereq = new RecognizeVerificationcodeRequest({});
        rpc_util_1.default.convert(request, recognizeVerificationcodereq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.imageURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        recognizeVerificationcodereq.imageURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let recognizeVerificationcodeResp = await this.recognizeVerificationcode(recognizeVerificationcodereq, runtime);
        return recognizeVerificationcodeResp;
    }
    async recognizePassportMRZ(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("RecognizePassportMRZ", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new RecognizePassportMRZResponse({}));
    }
    async recognizePassportMRZAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let recognizePassportMRZreq = new RecognizePassportMRZRequest({});
        rpc_util_1.default.convert(request, recognizePassportMRZreq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.imageURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        recognizePassportMRZreq.imageURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let recognizePassportMRZResp = await this.recognizePassportMRZ(recognizePassportMRZreq, runtime);
        return recognizePassportMRZResp;
    }
    async recognizeTakeoutOrder(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("RecognizeTakeoutOrder", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new RecognizeTakeoutOrderResponse({}));
    }
    async recognizeTakeoutOrderAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let recognizeTakeoutOrderreq = new RecognizeTakeoutOrderRequest({});
        rpc_util_1.default.convert(request, recognizeTakeoutOrderreq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.imageURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        recognizeTakeoutOrderreq.imageURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let recognizeTakeoutOrderResp = await this.recognizeTakeoutOrder(recognizeTakeoutOrderreq, runtime);
        return recognizeTakeoutOrderResp;
    }
    async recognizeQrCode(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("RecognizeQrCode", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new RecognizeQrCodeResponse({}));
    }
    async recognizeVATInvoice(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("RecognizeVATInvoice", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new RecognizeVATInvoiceResponse({}));
    }
    async recognizeVATInvoiceAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let recognizeVATInvoicereq = new RecognizeVATInvoiceRequest({});
        rpc_util_1.default.convert(request, recognizeVATInvoicereq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.fileURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        recognizeVATInvoicereq.fileURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let recognizeVATInvoiceResp = await this.recognizeVATInvoice(recognizeVATInvoicereq, runtime);
        return recognizeVATInvoiceResp;
    }
    async recognizeCharacter(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("RecognizeCharacter", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new RecognizeCharacterResponse({}));
    }
    async recognizeCharacterAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let recognizeCharacterreq = new RecognizeCharacterRequest({});
        rpc_util_1.default.convert(request, recognizeCharacterreq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.imageURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        recognizeCharacterreq.imageURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let recognizeCharacterResp = await this.recognizeCharacter(recognizeCharacterreq, runtime);
        return recognizeCharacterResp;
    }
    async recognizeTaxiInvoice(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("RecognizeTaxiInvoice", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new RecognizeTaxiInvoiceResponse({}));
    }
    async recognizeTaxiInvoiceAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let recognizeTaxiInvoicereq = new RecognizeTaxiInvoiceRequest({});
        rpc_util_1.default.convert(request, recognizeTaxiInvoicereq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.imageURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        recognizeTaxiInvoicereq.imageURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let recognizeTaxiInvoiceResp = await this.recognizeTaxiInvoice(recognizeTaxiInvoicereq, runtime);
        return recognizeTaxiInvoiceResp;
    }
    async recognizeIdentityCard(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("RecognizeIdentityCard", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new RecognizeIdentityCardResponse({}));
    }
    async recognizeIdentityCardAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let recognizeIdentityCardreq = new RecognizeIdentityCardRequest({});
        rpc_util_1.default.convert(request, recognizeIdentityCardreq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.imageURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        recognizeIdentityCardreq.imageURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let recognizeIdentityCardResp = await this.recognizeIdentityCard(recognizeIdentityCardreq, runtime);
        return recognizeIdentityCardResp;
    }
    async recognizeLicensePlate(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("RecognizeLicensePlate", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new RecognizeLicensePlateResponse({}));
    }
    async recognizeLicensePlateAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let recognizeLicensePlatereq = new RecognizeLicensePlateRequest({});
        rpc_util_1.default.convert(request, recognizeLicensePlatereq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.imageURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        recognizeLicensePlatereq.imageURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let recognizeLicensePlateResp = await this.recognizeLicensePlate(recognizeLicensePlatereq, runtime);
        return recognizeLicensePlateResp;
    }
    async recognizeTable(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("RecognizeTable", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new RecognizeTableResponse({}));
    }
    async recognizeTableAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let recognizeTablereq = new RecognizeTableRequest({});
        rpc_util_1.default.convert(request, recognizeTablereq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.imageURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        recognizeTablereq.imageURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let recognizeTableResp = await this.recognizeTable(recognizeTablereq, runtime);
        return recognizeTableResp;
    }
    async recognizeDrivingLicense(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("RecognizeDrivingLicense", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new RecognizeDrivingLicenseResponse({}));
    }
    async recognizeDrivingLicenseAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let recognizeDrivingLicensereq = new RecognizeDrivingLicenseRequest({});
        rpc_util_1.default.convert(request, recognizeDrivingLicensereq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.imageURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        recognizeDrivingLicensereq.imageURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let recognizeDrivingLicenseResp = await this.recognizeDrivingLicense(recognizeDrivingLicensereq, runtime);
        return recognizeDrivingLicenseResp;
    }
    async recognizeBankCard(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("RecognizeBankCard", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new RecognizeBankCardResponse({}));
    }
    async recognizeBankCardAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let recognizeBankCardreq = new RecognizeBankCardRequest({});
        rpc_util_1.default.convert(request, recognizeBankCardreq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.imageURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        recognizeBankCardreq.imageURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let recognizeBankCardResp = await this.recognizeBankCard(recognizeBankCardreq, runtime);
        return recognizeBankCardResp;
    }
    async recognizeTrainTicket(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("RecognizeTrainTicket", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new RecognizeTrainTicketResponse({}));
    }
    async recognizeTrainTicketAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let recognizeTrainTicketreq = new RecognizeTrainTicketRequest({});
        rpc_util_1.default.convert(request, recognizeTrainTicketreq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.imageURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        recognizeTrainTicketreq.imageURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let recognizeTrainTicketResp = await this.recognizeTrainTicket(recognizeTrainTicketreq, runtime);
        return recognizeTrainTicketResp;
    }
    async recognizeDriverLicense(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("RecognizeDriverLicense", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new RecognizeDriverLicenseResponse({}));
    }
    async recognizeDriverLicenseAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let recognizeDriverLicensereq = new RecognizeDriverLicenseRequest({});
        rpc_util_1.default.convert(request, recognizeDriverLicensereq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.imageURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        recognizeDriverLicensereq.imageURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let recognizeDriverLicenseResp = await this.recognizeDriverLicense(recognizeDriverLicensereq, runtime);
        return recognizeDriverLicenseResp;
    }
    async recognizeAccountPage(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("RecognizeAccountPage", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new RecognizeAccountPageResponse({}));
    }
    async recognizeAccountPageAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let recognizeAccountPagereq = new RecognizeAccountPageRequest({});
        rpc_util_1.default.convert(request, recognizeAccountPagereq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.imageURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        recognizeAccountPagereq.imageURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let recognizeAccountPageResp = await this.recognizeAccountPage(recognizeAccountPagereq, runtime);
        return recognizeAccountPageResp;
    }
    async recognizeStamp(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("RecognizeStamp", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new RecognizeStampResponse({}));
    }
    async recognizeStampAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let recognizeStampreq = new RecognizeStampRequest({});
        rpc_util_1.default.convert(request, recognizeStampreq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.imageURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        recognizeStampreq.imageURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let recognizeStampResp = await this.recognizeStamp(recognizeStampreq, runtime);
        return recognizeStampResp;
    }
    async recognizeBusinessCard(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("RecognizeBusinessCard", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new RecognizeBusinessCardResponse({}));
    }
    async recognizeBusinessCardAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let recognizeBusinessCardreq = new RecognizeBusinessCardRequest({});
        rpc_util_1.default.convert(request, recognizeBusinessCardreq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.imageURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        recognizeBusinessCardreq.imageURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let recognizeBusinessCardResp = await this.recognizeBusinessCard(recognizeBusinessCardreq, runtime);
        return recognizeBusinessCardResp;
    }
    async recognizeVINCode(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("RecognizeVINCode", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new RecognizeVINCodeResponse({}));
    }
    async recognizeVINCodeAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let recognizeVINCodereq = new RecognizeVINCodeRequest({});
        rpc_util_1.default.convert(request, recognizeVINCodereq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.imageURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        recognizeVINCodereq.imageURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let recognizeVINCodeResp = await this.recognizeVINCode(recognizeVINCodereq, runtime);
        return recognizeVINCodeResp;
    }
    async recognizeBusinessLicense(request, runtime) {
        tea_util_1.default.validateModel(request);
        return $tea.cast(await this.doRequest("RecognizeBusinessLicense", "HTTPS", "POST", "2019-12-30", "AK", null, $tea.toMap(request), runtime), new RecognizeBusinessLicenseResponse({}));
    }
    async recognizeBusinessLicenseAdvance(request, runtime) {
        // Step 0: init client
        let accessKeyId = await this._credential.getAccessKeyId();
        let accessKeySecret = await this._credential.getAccessKeySecret();
        let authConfig = new $RPC.Config({
            accessKeyId: accessKeyId,
            accessKeySecret: accessKeySecret,
            type: "access_key",
            endpoint: "openplatform.aliyuncs.com",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let authClient = new openplatform20191219_1.default(authConfig);
        let authRequest = new $OpenPlatform.AuthorizeFileUploadRequest({
            product: "ocr",
            regionId: this._regionId,
        });
        let authResponse = new $OpenPlatform.AuthorizeFileUploadResponse({});
        let ossConfig = new $OSS.Config({
            accessKeySecret: accessKeySecret,
            type: "access_key",
            protocol: this._protocol,
            regionId: this._regionId,
        });
        let ossClient = null;
        let fileObj = new $FileForm.FileField({});
        let ossHeader = new $OSS.PostObjectRequestHeader({});
        let uploadRequest = new $OSS.PostObjectRequest({});
        let ossRuntime = new $OSSUtil.RuntimeOptions({});
        rpc_util_1.default.convert(runtime, ossRuntime);
        let recognizeBusinessLicensereq = new RecognizeBusinessLicenseRequest({});
        rpc_util_1.default.convert(request, recognizeBusinessLicensereq);
        authResponse = await authClient.authorizeFileUploadWithOptions(authRequest, runtime);
        ossConfig.accessKeyId = authResponse.accessKeyId;
        ossConfig.endpoint = rpc_util_1.default.getEndpoint(authResponse.endpoint, authResponse.useAccelerate, this._endpointType);
        ossClient = new oss_client_1.default(ossConfig);
        fileObj = new $FileForm.FileField({
            filename: authResponse.objectKey,
            content: request.imageURLObject,
            contentType: "",
        });
        ossHeader = new $OSS.PostObjectRequestHeader({
            accessKeyId: authResponse.accessKeyId,
            policy: authResponse.encodedPolicy,
            signature: authResponse.signature,
            key: authResponse.objectKey,
            file: fileObj,
            successActionStatus: "201",
        });
        uploadRequest = new $OSS.PostObjectRequest({
            bucketName: authResponse.bucket,
            header: ossHeader,
        });
        await ossClient.postObject(uploadRequest, ossRuntime);
        recognizeBusinessLicensereq.imageURL = `http://${authResponse.bucket}.${authResponse.endpoint}/${authResponse.objectKey}`;
        let recognizeBusinessLicenseResp = await this.recognizeBusinessLicense(recognizeBusinessLicensereq, runtime);
        return recognizeBusinessLicenseResp;
    }
    getEndpoint(productId, regionId, endpointRule, network, suffix, endpointMap, endpoint) {
        if (!tea_util_1.default.empty(endpoint)) {
            return endpoint;
        }
        if (!tea_util_1.default.isUnset(endpointMap) && !tea_util_1.default.empty(endpointMap[regionId])) {
            return endpointMap[regionId];
        }
        return endpoint_util_1.default.getEndpointRules(productId, regionId, endpointRule, network, suffix);
    }
}
exports.default = Client;
